src/libraries/System.Drawing.Primitives/src/System/Drawing/KnownColorTable.cs

```diff
@@ -490,19 +499,50 @@ public static uint KnownColorToArgb(KnownColor color)
                  : ColorValueTable[(int)color];
         }
 
+        private static uint GetAlternateSystemColorArgb(KnownColor color)
+        {
+            // Shift the original (split) index to fit the alternate color map.
+            int index = color <= KnownColor.WindowText
+                ? (int)color
+                : (int)color - (int)KnownColor.ButtonFace + (int)KnownColor.WindowText + 1;
+
+            return AlternateSystemColors[index];
+        }
+
 #if FEATURE_WINDOWS_SYSTEM_COLORS
         public static uint GetSystemColorArgb(KnownColor color)
         {
             Debug.Assert(Color.IsKnownColorSystem(color));
 
-            return ColorTranslator.COLORREFToARGB(Interop.User32.GetSysColor((byte)ColorValueTable[(int)color]));
+            return !SystemColors.s_useAlternativeColorSet || HighContrastEnabled()
```

Comment:
Is this on a hot enough such that folks might notice an added PInvoke to get the high contrast setting?  IOW - should the value be cached?

Reply:
It needs to change on the fly or we need to start a window again to listen for system changes. I expect the cost to be trivial with most of it borne when you want alternative colors.

Reply:
I see, I suspected so.  Might be worthwhile to just double-check the perf since you'll be calling it *a ton*.